### ISSUE TITLE
100 related items being deleted from m2m relationship

### ISSUE THREAD

**Issue by u12206050** [directus/directus#15216]
### Describe the Bug

I have a m2m relationship between `persons` and `orgs` called `members`
On the `members` table I also have a field `status`.

When I create a new person and attach an org then save, all works well.

However when I create a new person, attach an org and edit the status field then the person is created, however a 100 other persons have their member record deleted.

### To Reproduce

[Directus_PivotM2M_issue.gz](https://github.com/directus/directus/files/9400870/Directus_PivotM2M_issue.gz)

Here is a database dump. 

Basically you need two collections with a m2m between them. On the m2m table add an additional field.
This also only happens when the attached item has more then 101 items. It deletes everything upto the last 100.

Check out how many items are in the DB for the org that you will attach before.
Create a new person item, attach a org item that already has more than 100 other persons, alter the field that is on the junction collection and save.
Check out how many items are in the DB for the org after.

### Errors Shown

No errors, just a ton of missing data

ALL FAKE GENERATED DATA

https://user-images.githubusercontent.com/6641242/186114329-b5961c99-37e2-434c-8168-55200994d454.mp4

.

### What version of Directus are you using?

9.16.1

### What version of Node.js are you using?

16.4.0

### What database are you using?

mysql 8

### What browser are you using?

chromium

### How are you deploying Directus?

Locally & GCP

**Comment by br41nslug** [comment]
So far i have been unable to get the database dump working properly or to reproduce it manually with the given instructions.

**Comment by emilyf** [comment]
It's hard to follow what's going on here but a few things that didn't make sense to me:

- Do you have a custom extension or any custom Vue code in your Directus codebase? Your console is outputting multiple invalid props
<img width="485" alt="Screen Shot 2022-08-23 at 12 51 22 PM" src="https://user-images.githubusercontent.com/1789807/186242222-e982a31c-41cd-44e6-af35-97aaf5740297.png">

- is your "persons" table a named alias for the Directus Users table or are you using your own custom table for that?

- When you're attaching an org to a person it seems like strange behaviors. Why are there TWO status fields on an org? It looks like you have the Directus system status field and a custom field? My understanding is if you set the core status field (which you do in the video) that performs a soft delete from the system.
<img width="717" alt="Screen Shot 2022-08-23 at 12 46 34 PM" src="https://user-images.githubusercontent.com/1789807/186243090-fdcbb97e-57f8-4eda-a612-ed4e040ddfab.png">

- When you're attaching an org to a person if you're choosing from a dropdown then from there why is it taking you to a CREATE org page after that? Note that you seem to end up with an org id of 47091 however in fact in your db the primary key seems to have every org with a primary key of 1
<img width="368" alt="Screen Shot 2022-08-23 at 1 00 30 PM" src="https://user-images.githubusercontent.com/1789807/186244216-dfab44ed-86af-480e-ad4c-31313a7b1164.png">
<img width="508" alt="Screen Shot 2022-08-23 at 1 01 07 PM" src="https://user-images.githubusercontent.com/1789807/186244288-562a8558-bb52-4a55-989b-78aea257c475.png">
- related to above, you're then checking it on the person record; I don't know what the goal is here but if you check a box selecting the record then you shouldn't have to select it again on the person screen; it should be selected
<img width="596" alt="Screen Shot 2022-08-23 at 12 50 57 PM" src="https://user-images.githubusercontent.com/1789807/186245867-b35971e8-6339-4057-b7b3-97ba8d69d6da.png">
<img width="598" alt="Screen Shot 2022-08-23 at 12 50 27 PM" src="https://user-images.githubusercontent.com/1789807/186245869-2456aa35-a56f-4e54-9d21-334c7ef11256.png">


- Why are you manually setting your person id instead of using an auto increment primaryKey? If you need an id specific to some other system you can use external identifer field on the user or another custom field.....so if you're setting the id then what is the type of that field as your first console output is saying it's getting an unexpected prop
- What is the primary key field on a person type? It seems like you're using "+" somehow based on the output? In which case it's not a primary key and so any others with that "+" would be affected too
<img width="461" alt="Screen Shot 2022-08-23 at 12 42 07 PM" src="https://user-images.githubusercontent.com/1789807/186246863-a27bcc55-a049-4a8e-a5dd-27ccb02b4c55.png">

- what's the purpose of this NO request?
<img width="147" alt="Screen Shot 2022-08-23 at 1 18 03 PM" src="https://user-images.githubusercontent.com/1789807/186247207-72c1d547-7233-4170-a144-e7991c9b68b9.png">

Not sure if any of this will solve your issue but it seems like there's definitely a problem with your relationships somewhere or perhaps a soft delete related to misuse of a primaryKey field override? Just some thoughts, hope one of them helps....

**Comment by u12206050** [comment]
@emilyf  Thank you for the effort, however there are a number of things you misunderstand:

> Do you have a custom extension or any custom Vue code in your Directus codebase? Your console is outputting multiple invalid props

No, I took the database dump together with a new Directus instance.

> is your "persons" table a named alias for the Directus Users table or are you using your own custom table for that?

It is a custom `persons` collection

> When you're attaching an org to a person it seems like strange behaviors. Why are there TWO status fields on an org? It looks like you have the Directus system status field and a custom field? My understanding is if you set the core status field (which you do in the video) that performs a soft delete from the system.

The second status field is a field on the junction table, not on the org. I am also only setting the second status field and not touching any fields on the org item itself.

> When you're attaching an org to a person if you're choosing from a dropdown then from there why is it taking you to a CREATE org page after that? Note that you seem to end up with an org id of 47091 however in fact in your db the primary key seems to have every org with a primary key of 1

47091 is the id for the junction row (the actual m2m relationship record)

> related to above, you're then checking it on the person record; I don't know what the goal is here but if you check a box selecting the record then you shouldn't have to select it again on the person screen; it should be selected

I wasn't checking it, I was selecting it so that I could "edit/set" the status field which is being displayed as part of the template on that relationship interface

> Why are you manually setting your person id instead of using an auto increment primaryKey? If you need an id specific to some other system you can use external identifer field on the user or another custom field.....so if you're setting the id then what is the type of that field as your first console output is saying it's getting an unexpected prop

This is not related to the issue, however the reason is simple, we set the id to an already defined external id number so that is why we set it manually.

> What is the primary key field on a person type? It seems like you're using "+" somehow based on the output? In which case it's not a primary key and so any others with that "+" would be affected too

This is how directus defines the payload when creating an item.

> what's the purpose of this NO request?

That is just a different collection item, specifically NO for Norway

Answering these questions I realise that this issue is complex, and does require a more than general in depth knowledge of how Directus works. Thanks though for trying.

**Comment by u12206050** [comment]
> So far i have been unable to get the database dump working properly or to reproduce it manually with the given instructions.

@br41nslug Did you try with mysql? Where did you get stuck?

**Comment by u12206050** [comment]
Update: The junction field itself doesn't have to be there. Any update caused within the drawer causes the full related object to be included in the payload and thus updated.

Since the array of related items are limited to 100 all other items will be deleted. Luckily for me, due to another "bug" it only deletes the first 100, since that is the default read limit.
